meson.build

```diff
@@ -21,6 +21,7 @@ mako_inc = include_directories('include')
 cc = meson.get_compiler('c')
 
 cairo = dependency('cairo')
+epoll = dependency('epoll-shim', required: false)
```

Comment:
Can we do something like wlroots here?

```meson
if host_machine.system() == 'freebsd'
	libepoll = dependency('epoll-shim')
else
	libepoll = dependency('', required: false)
endif
```

This avoids the check on non-FreeBSD, and fails the build if missing on FreeBSD, making it clearer it's not an optional dependency.

Reply:
epoll-shim is required on DragonFly, FreeBSD, NetBSD, OpenBSD and probably forks like MirBSD, MidnightBSD, GNU/kFreeBSD. However, only DragonFly and FreeBSD currently package/support wlroots compositors. DragonFly would need to port basu before noticing missing epoll-shim dependency in mako.

Some of those may follow IllumOS lead by moving epoll emulation into the kernel. I'd prefer to avoid platform whitelists which quickly grow out of date.

https://illumos.org/man/5/epoll
http://src.illumos.org/source/search?defs=epoll&project=illumos-gate

Reply:
I agree platform checks aren't great. Maybe replace with something like `cc.has_function('epoll')`?